internal/servicehash/servicehash.go

```diff
@@ -0,0 +1,56 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package servicehash
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"sync"
+)
+
+const (
+	maxCacheSize = 1000
+)
+
+var (
+	cache = make(map[string]string)
+	lock  sync.RWMutex
+)
+
+// Hash computes hash for `service`. Will use a local cache to avoid recomputation.
+func Hash(service string) string {
+	hash := getHashFromCache(service)
+	if len(hash) > 0 {
+		return hash
+	}
+	hashb := sha256.Sum256([]byte(service))
+	// Only grab first 10 characters
+	hash = hex.EncodeToString(hashb[:5])
```

Comment:
Comment says "first 10 characters" but the code is `[:5]`. Is one of these incorrect or am I misunderstanding this?

Reply:
👍 I was worried this comment wasn't quite clear enough :P Yeah since `hashb` is a byte array each byte is expanded to two hex characters in the string representation. How about adding something like  `2 chars per byte * 5 bytes = 10 chars` to the comment to make it more clear?

Reply:
Yeah that's more descriptive! 👍